themes/default/content/blog/pulumi-targets-dotnet-6/index.md

```diff
@@ -0,0 +1,182 @@
+---
+title: "Pulumi improved experience with .NET 6"
+authors: ["zaid-ajaj"]
+tags: ["dotnet", "csharp", "fsharp", "vb.net"]
+meta_desc: "Talking about the improved experience with Pulumi projects running on .NET 6, how it simplifies program structure and upgrades generated C# code."
+meta_image: thumbnail.png
+date: "2022-07-21"
+---
+
+In this blog post, we will talk about how Pulumi is now using [.NET 6](https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-6), the latest Long-Term Support version of .NET as our default across the ecosystem. We will discuss the changes applied to templates, program structure and code generation. We also explain how Pulumi C# projects can benefit from latest features in .NET 6 and how it simplifies your programs overall. Let's dive in, shall we?
+
+<!--more-->
+
+### Templates
+
+All of our .NET templates have been updated which now use `net6.0` as the target framework. This means that projects created with `pulumi new` require at least the [.NET 6 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) to build and run the projects. The templates also simplified the program structure quite a bit and not longer use `MyStack` classes to define resources in case of C# and VB.NET templates. F# templates didn't change that much because these already didn't use the class approach to define infrastructure.
```

Comment:
> All of our .NET templates have been updated which now use `net6.0` as the target framework.

For what version of Pulumi? When was this released?

```suggestion
All of our .NET templates have been updated which now use `net6.0` as the target framework. This means that projects created with `pulumi new` require at least the [.NET 6 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) to build and run the projects. The templates also simplified the program structure quite a bit and not longer use `MyStack` classes to define resources in case of C# and VB.NET templates. F# templates didn't change much because these already didn't use the class approach to define infrastructure.
```

Reply:
Templates are not versioned because they are retrieved from `pulumi/templates` by any Pulumi CLI version